src/classes/Carts/UserCart.ts

```diff
@@ -415,8 +415,8 @@ export default class UserCart extends Cart {
                         'I only have ' +
                         pluralize(this.bot.schema.getName(SKU.fromString(sku), false), ourAssetids.length, true);
 
-                    // Add the max amount to the cart
-                    this.addOurItem(sku, amount);
+                    // Add the max amount to the cart and substract added amount from command (if exist)
+                    this.addOurItem(sku, this.our[sku] ? amount - this.our[sku] : amount);
```

Comment:
This line shouldn't change because it is always only adding amount which is equal to the total amount of assets we have or the amount requested.